pyfar/signal.py

```diff
@@ -118,60 +47,126 @@ def domain(self, new_domain):
         if new_domain not in self._VALID_SIGNAL_DOMAIN:
             raise ValueError("Incorrect domain, needs to be time/freq.")
 
-        if not (self._domain == new_domain):
-            # Only process if we change domain
-            if new_domain == 'time':
-                # If the new domain should be time, we had a saved spectrum
-                # and need to do an inverse Fourier Transform
-                self.time = fft.irfft(
-                    self._data, self.n_samples, self._sampling_rate,
-                    self._fft_norm)
-            elif new_domain == 'freq':
-                # If the new domain should be freq, we had sampled time data
-                # and need to do a Fourier Transform
-                self.freq = fft.rfft(
-                    self._data, self.n_samples, self._sampling_rate,
-                    self._fft_norm)
+        self._domain = new_domain
```

Comment:
I would remove the domain setter here. The domain setter is only meaningful for Signal. All other classes either do not have a domain, or are limited to a single domain. For example changing the domain on the time domain class would require a cast to a different container class (and a respective ifft)

Reply:
Good point - removed the setter from `Audio`. It is now only available for `Signal`